spec/text-editor-component-spec.js

```diff
@@ -411,26 +411,28 @@ describe('TextEditorComponent', () => {
       await component.getNextUpdatePromise()
       const [cursor1, cursor2] = element.querySelectorAll('.cursor')
 
-      expect(getComputedStyle(cursor1).opacity).toBe('1')
-      expect(getComputedStyle(cursor2).opacity).toBe('1')
-
-      await conditionPromise(() =>
-        getComputedStyle(cursor1).opacity === '0' && getComputedStyle(cursor2).opacity === '0'
-      )
-
       await conditionPromise(() =>
         getComputedStyle(cursor1).opacity === '1' && getComputedStyle(cursor2).opacity === '1'
       )
-
       await conditionPromise(() =>
         getComputedStyle(cursor1).opacity === '0' && getComputedStyle(cursor2).opacity === '0'
       )
+      await conditionPromise(() =>
+        getComputedStyle(cursor1).opacity === '1' && getComputedStyle(cursor2).opacity === '1'
+      )
 
       editor.moveRight()
       await component.getNextUpdatePromise()
 
-      expect(getComputedStyle(cursor1).opacity).toBe('1')
-      expect(getComputedStyle(cursor2).opacity).toBe('1')
```

Comment:
@nathansobo and I discussed this ☝️ today. Summary:

- We think that the assertion changes are appropriate for the [first batch of operations](https://github.com/atom/atom/blob/29810c6cd1924e08126bd18d759b80abfd1c199c/spec/text-editor-component-spec.js#L408-L412) in this test. We don't offer any guarantees about the _initial_ cursor visibility state following that set of operations.
- We think the assertion changes are not needed for the [second batch of operations](https://github.com/atom/atom/blob/29810c6cd1924e08126bd18d759b80abfd1c199c/spec/text-editor-component-spec.js#L424) in this test. After moving the cursor, the cursor should indeed be in the visible state, so the original assertions should allow us to verify that behavior.